eng/native/configuretools.cmake

```
@@ -41,10 +41,9 @@ if(NOT WIN32 AND NOT CLR_CMAKE_TARGET_BROWSER)
 
   locate_toolchain_exec(ar CMAKE_AR)
   locate_toolchain_exec(nm CMAKE_NM)
+  locate_toolchain_exec(ranlib CMAKE_RANLIB)
 
-  if(CMAKE_C_COMPILER_ID MATCHES "GNU")
-    locate_toolchain_exec(ranlib CMAKE_RANLIB)
-  else()
+  if(NOT CMAKE_C_COMPILER_ID MATCHES "GNU")
```

nit: we are not using `NOT` pattern with `CMAKE_C{XX}_COMPILER_ID` in other places
```suggestion
  if(CMAKE_C_COMPILER_ID STREQUAL Clang)
```

---

Thanks for the suggestion, I've replaced it with `if(CMAKE_C_COMPILER_ID MATCHES "Clang")` as we use the same in the code above.